workflow/metrics/test_exporter.go

```diff
@@ -139,6 +139,23 @@ func (t *TestExporter) getNamedFloat64GaugeData(name string, attribs *attribute.
 	return nil, fmt.Errorf("%s type gauge[float64] not found in %v", name, mtc)
 }
 
+func (t *TestExporter) getNamedInt64GaugeData(name string, attribs *attribute.Set) (*metricdata.DataPoint[int64], error) {
+	mtc, err := t.getNamedMetric(name)
+	if err != nil {
+		return nil, err
+	}
+
+	if gauge, ok := mtc.Data.(metricdata.Gauge[int64]); ok {
+		for _, dataPoint := range gauge.DataPoints {
+			if dataPoint.Attributes.Equals(attribs) {
+				return &dataPoint, nil
+			}
+		}
+		return nil, fmt.Errorf("%s named gauge[float64] with attribs %v not found in %v", name, attribs, mtc)
+	}
+	return nil, fmt.Errorf("%s type gauge[float64] not found in %v", name, mtc)
```

Comment:
similarly could invert the first conditional (which I believe is how it's usually done in Go):

```suggestion
	gauge, ok := mtc.Data.(metricdata.Gauge[int64])
	if !ok {
		return nil, fmt.Errorf("%s type gauge[float64] not found in %v", name, mtc)
	}

	for _, dataPoint := range gauge.DataPoints {
		if dataPoint.Attributes.Equals(attribs) {
			return &dataPoint, nil
		}
	}
	return nil, fmt.Errorf("%s named gauge[float64] with attribs %v not found in %v", name, attribs, mtc)
```